llvm/include/llvm/CodeGen/MachineMemOperand.h

```diff
@@ -164,10 +164,13 @@ class MachineMemOperand {
     unsigned SSID : 8;            // SyncScope::ID
     /// Atomic ordering requirements for this memory operation. For cmpxchg
     /// atomic operations, atomic ordering requirements when store occurs.
-    unsigned Ordering : 4;        // enum AtomicOrdering
+    unsigned Ordering : 3; // enum AtomicOrdering
     /// For cmpxchg atomic operations, atomic ordering requirements when store
     /// does not occur.
-    unsigned FailureOrdering : 4; // enum AtomicOrdering
+    unsigned FailureOrdering : 3; // enum AtomicOrdering
+    /// Whether capability comparisons are exact for cmpxchg atomic operations.
+    unsigned ExactCompare : 1;
```

Comment:
I considered adding two new ISD:: operations for the exact CMPXCHG, but adding it here seems less invasive.